### PR TITLE
Fix ks-controller panic due to missing nil checking in mutli-branch pipeline

### LINKED ISSUE #3176: ks-controller panic due to missing nil checking
**Describe the Bug**

Exception stack:
```
E1203 01:59:21.920515 1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 1274 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x1f697a0, 0x3a00380)
/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/util/runtime/runtime.go:74 +0xa3
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/util/runtime/runtime.go:48 +0x82
panic(0x1f697a0, 0x3a00380)
/opt/hostedtoolcache/go/1.13.15/x64/src/runtime/panic.go:679 +0x1b2
kubesphere.io/kubesphere/pkg/simple/client/devops/jenkins/internal.AppendGitlabSourceToEtree(0xc003d01080, 0x0)
/home/runner/work/kubesphere/kubesphere/pkg/simple/client/devops/jenkins/internal/gitlab.go:13 +0xd8
kubesphere.io/kubesphere/pkg/simple/client/devops/jenkins.createMultiBranchPipelineConfigXml(0xc0015c9ec0, 0xe, 0xc002e10a20, 0xc003d1ee01, 0xc0030f1a10, 0x1, 0x1)
/home/runner/work/kubesphere/kubesphere/pkg/simple/client/devops/jenkins/pipeline_internal.go:821 +0x88a
kubesphere.io/kubesphere/pkg/simple/client/devops/jenkins.(*Jenkins).CreateProjectPipeline(0xc0007000f0, 0xc0015c9ec0, 0xe, 0xc000f3b680, 0x6, 0x0, 0x2707d80, 0xc003d1ee20)
/home/runner/work/kubesphere/kubesphere/pkg/simple/client/devops/jenkins/project_pipeline.go:54 +0x21a
kubesphere.io/kubesphere/pkg/controller/pipeline.(*Controller).syncHandler(0xc000698be0, 0xc0015c9ec0, 0x15, 0xc0018bb560, 0xc000ca9cf0)
/home/runner/work/kubesphere/kubesphere/pkg/controller/pipeline/pipeline_controller.go:252 +0xae0
kubesphere.io/kubesphere/pkg/controller/pipeline.(*Controller).processNextWorkItem.func1(0xc000698be0, 0x1e94b80, 0xc000f85120, 0x0, 0x0)
/home/runner/work/kubesphere/kubesphere/pkg/controller/pipeline/pipeline_controller.go:144 +0xf3
kubesphere.io/kubesphere/pkg/controller/pipeline.(*Controller).processNextWorkItem(0xc000698be0, 0x1)
/home/runner/work/kubesphere/kubesphere/pkg/controller/pipeline/pipeline_controller.go:151 +0x51
kubesphere.io/kubesphere/pkg/controller/pipeline.(*Controller).worker(0xc000698be0)
/home/runner/work/kubesphere/kubesphere/pkg/controller/pipeline/pipeline_controller.go:164 +0x2b
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc000cfc350)
/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/util/wait/wait.go:152 +0x5e
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000cfc350, 0x3b9aca00, 0x0, 0x1, 0xc00089e000)
/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(0xc000cfc350, 0x3b9aca00, 0xc00089e000)
/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/util/wait/wait.go:88 +0x4d
created by kubesphere.io/kubesphere/pkg/controller/pipeline.(*Controller).Run
/home/runner/work/kubesphere/kubesphere/pkg/controller/pipeline/pipeline_controller.go:184 +0x23f
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1c3f0a8]
```

**Versions Used**
KubeSphere: 3.0

`kubespheredev/ks-controller-manager:latest@sha256:64ed223688d705ce77991b288018cc72e370dc1b0b89d09f067de69021af6767`

**Environment**
None.

**How To Reproduce**
None

**Expected behavior**
Any errors should let ks-controller-manager panic.

### PR DESCRIPTION
**Pull request by LinuxSuRen**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
This issue makes ks-controller-manager panic.

**Which issue(s) this PR fixes**:
Fixes #3176

**Special notes for reviewers**:

**Additional documentation, usage docs, etc.**:

### CODE REVIEW

File: pkg/simple/client/devops/jenkins/internal/pull_request.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
+package internal
+
+type PRDiscoverTrust int
+
+const (
+	PRDiscoverTrustMember     PRDiscoverTrust = 1
+	PRDiscoverTrustEveryone   PRDiscoverTrust = 2
+	PRDiscoverTrustPermission PRDiscoverTrust = 3
+	PRDiscoverTrustNobody     PRDiscoverTrust = 4
+)
```
Comment (shaowenchen) [inline pkg/simple/client/devops/jenkins/internal/pull_request.go:10]: Please define -1 here as a default value.

File: pkg/simple/client/devops/jenkins/internal/github.go [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+		if githubSource.CloneOption.Depth >= 0 {
+			cloneExtension.CreateElement("depth").SetText(strconv.Itoa(githubSource.CloneOption.Depth))
+		} else {
+			cloneExtension.CreateElement("depth").SetText(strconv.Itoa(1))
+		}
+	}
+	if githubSource.RegexFilter != "" {
+		regexTraits := traits.CreateElement("jenkins.scm.impl.trait.RegexSCMHeadFilterTrait")
+		regexTraits.CreateAttr("plugin", "scm-api@2.4.0")
```
Comment (shaowenchen) [inline pkg/simple/client/devops/jenkins/internal/github.go:69]: Why does it need a hard code version? If another version, will there be a problem?

File: pkg/simple/client/devops/jenkins/internal/bitbucket.go [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+		if gitSource.CloneOption.Depth >= 0 {
+			cloneExtension.CreateElement("depth").SetText(strconv.Itoa(gitSource.CloneOption.Depth))
+		} else {
+			cloneExtension.CreateElement("depth").SetText(strconv.Itoa(1))
+		}
+	}
+	if gitSource.RegexFilter != "" {
+		regexTraits := traits.CreateElement("jenkins.scm.impl.trait.RegexSCMHeadFilterTrait")
+		regexTraits.CreateAttr("plugin", "scm-api@2.4.0")
```
Comment (shaowenchen) [inline pkg/simple/client/devops/jenkins/internal/bitbucket.go:74]: The same to another.

File: pkg/simple/client/devops/jenkins/internal/github.go (hunk 2 above)
Comment (LinuxSuRen) [reply pkg/simple/client/devops/jenkins/internal/github.go:69]: I just refractory these code lines. You can see it from the master branch. https://github.com/kubesphere/kubesphere/blob/91c2e05616ef7abcdb51bf774df02f8eed803030/pkg/simple/client/devops/jenkins/pipeline_internal.go#L291

I can do some tests to see it still works well after removing the specified version.

Comment (LinuxSuRen) [reply pkg/simple/client/devops/jenkins/internal/github.go]: It works well. See https://github.com/kubesphere/kubesphere/pull/3177/commits/e4782b37f0cd007952a8e466f3667bd4ba8a5496#diff-bc352fa9668f5879136b80d2ab0539b58c06a6988bdb8282c34878a845621d86R74

File: pkg/simple/client/devops/jenkins/internal/pull_request.go (hunk 1 above)
Comment (LinuxSuRen) [reply pkg/simple/client/devops/jenkins/internal/pull_request.go:11]: Ok. see https://github.com/kubesphere/kubesphere/pull/3177/commits/e4782b37f0cd007952a8e466f3667bd4ba8a5496#diff-59e0919f8ba6868f2244ad715cbd79940b091e7ca380681df5c3028787f168c0R48
https://github.com/kubesphere/kubesphere/pull/3177/commits/e4782b37f0cd007952a8e466f3667bd4ba8a5496#diff-59e0919f8ba6868f2244ad715cbd79940b091e7ca380681df5c3028787f168c0R10

File: pkg/simple/client/devops/jenkins/internal/bitbucket.go [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+	if gitSource.DiscoverPRFromOrigin != 0 {
+		traits.CreateElement("com.cloudbees.jenkins.plugins.bitbucket.OriginPullRequestDiscoveryTrait").
+			CreateElement("strategyId").SetText(strconv.Itoa(gitSource.DiscoverPRFromOrigin))
+	}
+	if gitSource.DiscoverPRFromForks != nil {
+		forkTrait := traits.CreateElement("com.cloudbees.jenkins.plugins.bitbucket.ForkPullRequestDiscoveryTrait")
+		forkTrait.CreateElement("strategyId").SetText(strconv.Itoa(gitSource.DiscoverPRFromForks.Strategy))
+		trustClass := "com.cloudbees.jenkins.plugins.bitbucket.ForkPullRequestDiscoveryTrait$"
+		switch gitSource.DiscoverPRFromForks.Trust {
+		case 1:
```
[the remark this answers is not in the document]
Comment (LinuxSuRen) [reply pkg/simple/client/devops/jenkins/internal/bitbucket.go]: Yes, I missed this part before. See https://github.com/kubesphere/kubesphere/pull/3177/commits/f50f30e6a352c0c876bc25a28485785c582ab665

**Comment by zryfish** [comment]
Are there more comments from @shaowenchen ?

**Comment by shaowenchen** [comment]
> Are there more comments from @shaowenchen ?

 /lgtm

**Pull request merged**